### DISCUSSION TITLE
Highlight keywords (TODO, FIX, ...)

### DISCUSSION THREAD

**Discussion by Mirodin** [orbitalquark/textadept#112] [Q&A] [answered]
Hey,

how do I make Textadept (v.11.1) highlight specific keywords i.e. when developing Lua modules? I grew quite fond of Neovim highlighting keywords like TODO and FIX in my comments so it is easier to find places that need attention.

[textadept.editing.highlight_words](https://orbitalquark.github.io/textadept/api.html#textadept.editing.highlight_words) is not what I am looking for since I want those words highlighted in all Lua, Python, etc. files but not in "normal" text files.

[ui.find.highlight_all_matches](https://orbitalquark.github.io/textadept/api.html#ui.find.highlight_all_matches) only works for words I search for via Strg+F.

**Comment by orbitalquark** [comment, marked as answer]
A quick and dirty way would be to put the following in your *~/.textadept/init.lua*:

    local TODO = _SCINTILLA.next_indic_number()
    view.indic_style[TODO] = view.INDIC_ROUNDBOX
    view.indic_fore[TODO] = 0x0000FF

    function highlight_todo(name)
      buffer.indicator_current = TODO
      buffer:indicator_clear_range(1, buffer.length)
      for pos in buffer:get_text():gmatch('()TODO') do
        if buffer.style_at[pos] == buffer:style_of_name('comment') then
          buffer:indicator_fill_range(pos, 4)
        end
      end
    end

    keys['alt+a'] = highlight_todo

It just searches through the buffer for any 'TODO' text in comments and highlights them with an indicator of your choosing. If you wanted to expand it to also highlight 'FIXME', you can add another `for in` loop.

You can try connecting `highlight_todo` to other events for a more hands-off approach, but that might be a bit expensive in terms of performance (e.g. on every keypress).

Another option is to modify language lexers to look for and style 'TODO' and 'FIXME' text in comments, but you'd have to do that for every lexer you're interested in.

**Reply by Mirodin** [reply]
Thank you very much! Works like a charm! :)
